server/src/main/java/io/envoyproxy/controlplane/server/DiscoveryServer.java

```diff
@@ -339,4 +377,79 @@ private void send(Response response, String typeUrl) {
       }
     }
   }
+
+  public static class Builder {
+
+    private final ConfigWatcher configWatcher;
+
+    private List<DiscoveryServerCallbacks> callbacks;
+
+    private ExecutorGroup executorGroup;
+
+    private ProtoResourcesSerializer protoResourcesSerializer;
+
+    private RequestLimiter requestLimiter;
+
+    private FlowControl.Factory<DiscoveryResponse> flowControlFactory;
+
+    private Builder(ConfigWatcher configWatcher) {
+      this.configWatcher = configWatcher;
+    }
+
+    /**
+     * Creates an instance of {@link DiscoveryServer} based on provided configuration merged with defaults.
+     * @return instance of {@link DiscoveryServer}
+     */
+    public DiscoveryServer build() {
+      List<DiscoveryServerCallbacks> callbacks = this.callbacks == null
+          ? Collections.emptyList()
```

Comment:
You can initialize the fields with emptyList instead of using these ternaries, and do similar default initialization of the other args

Reply:
I wanted to avoid unnecessary initialization of these objects, but they cost close to nothing, I'll do as you suggest.